### ISSUE TITLE
Animation with TimingCallBacks

### ISSUE THREAD

**Issue by furkanturan** [paulrosen/abcjs#271]
Hello,

I saw the recent commits to the examples directory. One of these examples is the use of cursor animation. It implements the cursor animations using the `startAnimation` like API calls. In contrast, the `api.md` says that these animation API calls are deprecated, and replaced with `TimingCallbacks`. However, the examples does not prefer to use them.

May I kindly ask which is the suggested way to implement animations?
Also is there any example on how use the `TimingCallbacks`?

Kind Regards,

**Comment by paulrosen** [comment]
Yes, `TimingCallbacks` is the preferred way to do animations. Sorry about the confusion. I am in the process of revamping the documentation, but don't have anything to show for it yet.

**Comment by furkanturan** [comment]
Thank you for clarification. I just wanted to make a basic example, that blinks the background of the page, at every beat. I suppose, it will be a simple tool to practice with, so that I can synchronise my playing with the beats. Also, it could be a simple example to show how animations work. If I manage to do it without an example, maybe I can commit into the examples directory.

**Comment by Tyutyesz** [comment]
I've asked this question on [stackoverflow ](https://stackoverflow.com/questions/56418769/no-moving-cursor-when-playing-the-abcjs-animation/59304866#59304866 )too, becouse i would liked to use it in **Angular 7+**. **Pauli** has a good answer for it, and I managed to have the Timing callbacks. I will use this for my own animation for the cursor. 
`timer = new abcjs.TimingCallbacks(this.mySheet[0], {
      beatCallback: this.beatCallbackIntercept,
      eventCallback: this.eventCallback,
      lineEndCallback: this.lineEndCallbackIntercept,
    })`

And implement a function which will start the **timer**:

`start() {
    this.timer.start();
  }`
In the `this.eventCallback` you can read all the parameters you will need for your animations:
`eventCallback(e){
  console.log(e);
}`

**Comment by furkanturan** [comment]
Hello, thanks for sharing your trick. In fact, I managed to use it as well, and as I promised in my above comment, I created an example and send a pull request with that. However, Paul Rosen has plans for updating the midi player, so didn't prefer to accept the commit. 

For me, with this old midi player everything works. In fact it is a very simple example. It only flashes the screen background while playing the song, so that I can synchronize easily with the beats, while I am practicing the song with my instrument.

Thanks again.
